connections/github.py

```diff
@@ -28,15 +30,23 @@ def get_token():
     with open(private_key_path, 'r') as private_key_file:
         private_key = private_key_file.read()
 
-    # If the config keys are not set, get_access_token will raise a NotImplementedError
-    # Returning NoneType token will stop the connection in get_instance
-    try:
-        github_integration = GithubIntegration(app_id, private_key)
-        # Note that installation access tokens last only for 1 hour, you will need to regenerate them after they expire.
-        _token = github_integration.get_access_token(installation_id)
-    except NotImplementedError as err:
-        logging.error(err)
-        _token = None
+    tries = 3
+    for i in range(tries):
+        # If the config keys are not set, get_access_token will raise a NotImplementedError
+        # Returning NoneType token will stop the connection in get_instance
+        try:
+            github_integration = GithubIntegration(app_id, private_key)
+            # Note that installation access tokens last only for 1 hour,
+            # you will need to regenerate them after they expire.
+            _token = github_integration.get_access_token(installation_id)
+        except NotImplementedError as err:
+            if i < tries - 1:
+                time.sleep(0.8)
```

Comment:
This always waits 0.8 seconds. Maybe the waiting time could be $n * (i+1)$ seconds? $n$ being the default/minimum waiting time (thus waiting time would increase linearly: $n, 2n, 3n, \dots$). Alternatively, the waiting time could be $n^{(i+1)}$ seconds (thus waiting time would increase exponentially: $n, n^2, n^3, \dots$).

Reply:
To be on the safe side, maybe wait for $1 + i * n$ seconds.